pymilo/transporters/preprocessing_transporter.py

```diff
@@ -0,0 +1,102 @@
+
+        serialize the data[key] of the given model which type is model_type.
+        basically in order to fully serialize a model, we should traverse over all the keys of its data dictionary and
+        pass it through the chain of associated transporters to get fully serialized.
+
+        :param data: the internal data dictionary of the given model
+        :type data: dict
+        :param key: the special key of the data param, which we're going to serialize its value(data[key])
+        :type key: object
+        :param model_type: the model type of the ML model, which data dictionary is given as the data param
+        :type model_type: str
+        :return: pymilo serialized output of data[key]
+        """
+        if self.is_preprocessing_module(data[key]):
+            return self.serialize_pre_module(data[key])
+        return data[key]
+
+
+    def deserialize(self, data, key, model_type):
+        """
+        Deserialize previously pymilo serialized preprocessing object.
+
+        deserialize the data[key] of the given model which type is model_type.
+        basically in order to fully deserialize a model, we should traverse over all the keys of its serialized data dictionary and
+        pass it through the chain of associated transporters to get fully deserialized.
+
+        :param data: the internal data dictionary of the associated json file of the ML model which is generated previously by
+        pymilo export.
+        :type data: dict
+        :param key: the special key of the data param, which we're going to deserialize its value(data[key])
+        :type key: object
+        :param model_type: the model type of the ML model, which internal serialized data dictionary is given as the data param
+        :type model_type: str
+        :return: pymilo deserialized output of data[key]
+        """
+        content = data[key]
+        if self.is_preprocessing_module(content):
+            return self.deserialize_pre_module(content)
+        return content
+
+
+    def is_preprocessing_module(self, pre_module):
+        """
+        Check whether the given module is a sklearn Preprocessing module or not.
+
+        :param pre_module: given object
+        :type pre_module: any
+        :return: bool
+        """
+        if isinstance(pre_module, dict):
+            return check_str_in_iterable(
+                "pymilo-preprocessing-type",
+                pre_module) and pre_module["pymilo-preprocessing-type"] in SKLEARN_PREPROCESSING_TABLE.keys()
+        return get_sklearn_type(pre_module) in SKLEARN_PREPROCESSING_TABLE.keys()
+
+
+    def serialize_pre_module(self, pre_module):
+        """
+        Serialize Preprocessing object.
+
+        :param pre_module: given sklearn preprocessing module
+        :type pre_module: sklearn.preprocessing
+        :return: pymilo serialized pre_module
+        """
+        gdst = GeneralDataStructureTransporter()
+        gdst.transport(pre_module, Command.SERIALIZE, False)
```

Comment:
Are these two lines necessary?

Reply:
Yes they are, `transport` function applies in-place changes to `pre_module` which is used later.